### PR TITLE
new package: "match-sorter"

### PR DESCRIPTION
**Pull request by claasahl**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by plantain-00 — APPROVED** [review]

File: types/match-sorter/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+/**
+ * Takes an array of items and a value and returns a new array with the items that match the given value
+ * @param items - the items to sort
+ * @param value - the value to use for ranking
+ * @param options - Some options to configure the sorter
+ * @return the new sorted array
+ */
+export default function matchSorter<T>(items: T[], value: string, options?: Options<T>): T[];
```
Comment (mhegazy) [inline types/match-sorter/index.d.ts]: Please see https://github.com/DefinitelyTyped/DefinitelyTyped#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default

**Review by mhegazy — CHANGES_REQUESTED** [review]

**Comment by mhegazy** [comment]
To be clear `export default function matchSorter` should instead be `export  = matchSorter`

**Comment by claasahl** [comment]
Thanks for the feedback, @mhegazy 

Does this also apply when the original library uses "export default"? (see
https://github.com/kentcdodds/match-sorter/blob/5b793831e9f174b923382d222b266c4d8010f3de/src/index.js#L431)

**Comment by mhegazy** [comment]
Looks like this code is transpiled..
this is what i see at runtime:

```shell
c:\test\dt>npm install match-sorter
+ match-sorter@2.2.3
updated 1 package and audited 2 packages in 3.938s

c:\test\dt>node
> var match = require("match-sorter");
undefined
> match
{ [Function: matchSorter]
  rankings:
   { CASE_SENSITIVE_EQUAL: 9,
     EQUAL: 8,
     STARTS_WITH: 7,
     WORD_STARTS_WITH: 6,
     STRING_CASE: 5,
     STRING_CASE_ACRONYM: 4,
     CONTAINS: 3,
     ACRONYM: 2,
     MATCHES: 1,
     NO_MATCH: 0 },
  caseRankings: { CAMEL: 0.8, PASCAL: 0.6, KEBAB: 0.4, SNAKE: 0.2, NO_CASE: 0 } }
>
```
match-sorter module is just a function with two properties.

**Comment by claasahl** [comment]
thanks for the clarification, @mhegazy 

I have pushed an updated version.

**Pull request merged**